dags/process_elife_zip_dag.py

```diff
@@ -45,65 +43,43 @@
-    zip_file_name = conf.get('file')
-    logger.info('FILE NAME PASSED FROM TRIGGER= %s', zip_file_name)
-    message = '%s triggered without a file name passed to conf' % dag_run.dag_id
-    assert zip_file_name, message
-
-    # get/validate article name
+def get_expected_elife_article_name(file_name: str) -> str:
     pattern = r'^\w+-\w+'
-    article_name = re.search(pattern, zip_file_name)
+    article_name = re.search(pattern, file_name)
     error_message = ('%s is malformed. Expected archive name to start with '
                      'any number/character, hyphen, any number/character (%s)'
-                     'example: name-id.extension' % (zip_file_name, pattern))
-    assert article_name and zip_file_name.startswith(article_name.group()), error_message
-    article_name = article_name.group() + '.xml'
+                     'example: name-id.extension' % (file_name, pattern))
+    assert article_name and file_name.startswith(
+        article_name.group()), error_message
+    return article_name.group() + '.xml'
 
-    # temporary files are securely stored on disk and automatically deleted when closed
-    with TemporaryFile() as temp_file:
+
+def extract_archived_files_to_bucket(**context):
+    zip_file_name = get_file_name_passed_to_dag_run_conf_file(**context)
+    article_name = get_expected_elife_article_name(zip_file_name)
+
+    with TemporaryFile(dir=TEMP_DIRECTORY) as temp_zip_file:
         s3 = get_aws_connection('s3')
-        # store downloaded file in temp file
-        s3.download_fileobj(SOURCE_BUCKET, zip_file_name, temp_file)
-        logger.info('ZIPPED FILES= %s', ZipFile(temp_file).namelist())
+        s3.download_fileobj(
+            Bucket=SOURCE_BUCKET,
+            Key=zip_file_name,
+            Fileobj=temp_zip_file
+        )
+        logger.info('ZIPPED FILES= %s', ZipFile(temp_zip_file).namelist())
 
-        # extract zip to cloud bucket
         folder_name = zip_file_name.rstrip('.zip')
-        file_to_upload = None
-
-        for zipped_file_path in ZipFile(temp_file).namelist():
-
-            if is_image(zipped_file_path) and not is_allowed(zipped_file_path):
-                # convert image to jpeg
-                file_to_upload = BytesIO()
-                image = Image.open(BytesIO(ZipFile(temp_file).read(zipped_file_path)))
-                try:
-                    image.save(file_to_upload, 'JPEG')
-                except IOError:
-                    image = image.convert('RGB')
-                    image.save(file_to_upload, 'JPEG')
-                file_to_upload = file_to_upload.getvalue()
-                zipped_file_path = re.sub(r'\.\w+$', '.jpg', zipped_file_path)
-
-            if is_allowed(zipped_file_path):
+
+        # extract zip to cloud bucket
+        for zipped_file_path in ZipFile(temp_zip_file).namelist():
+            with TemporaryFile(dir=TEMP_DIRECTORY) as temp_zipped_file:
+                temp_zipped_file.write(ZipFile(temp_zip_file).read(zipped_file_path))
```

Comment:
strange variable name: the file being written is not zipped anymore, `temp_expanded_file`|`temp_extracted_file`|`temp_unzipped_file`?

Reply:
I'll update to `temp_unzipped_file`